src/services/Api/contacts.ts

```diff
@@ -239,3 +239,113 @@ export async function updateContacts(
   }
 }
+
+export async function fetchDrafts(
+  token: string,
+  orgId: number,
+): Promise<LetterDraft[]> {
+  console.log('made req');
+  const requestOptions: RequestInit = {
+    method: 'GET',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${token}`,
+    },
+  };
+  const response = await fetch(
+    url.resolve(API_URL, 'org/' + orgId + '/drafts'),
+    requestOptions,
+  );
+  const body = await response.json();
+  if (body.status === 'ERROR') {
+    throw body;
+  }
+  var draftsData: LetterDraft[] = [];
+  console.log(body.data);
+  body.data.forEach(
+    (user: { user: RawDraftSender; drafts: RawLetterDraft[] }) => {
+      user.drafts.forEach((draft: RawLetterDraft) => {
+        console.log(draft);
+        const letterDraft: LetterDraft = {
+          user_id: user.user.id,
+          contact_id: draft.contact_id,
+          first_name: user.user.first_name,
+          last_name: user.user.last_name,
+          contact_name: draft.contact_name,
+          updated_at: draft.updated_at,
+          content: draft.content,
+        };
+        console.log(letterDraft);
+        draftsData = [...draftsData, letterDraft];
+      });
+    },
+  );
+  console.log(draftsData);
+  return draftsData;
+}
+
+export async function createDirectLetter(
+  token: string,
+  newsletter: DraftDirectLetter,
+): Promise<DirectMailLog> {
+  let formData = new FormData();
+  formData.append('type', 'pdf');
+  if (newsletter?.file) {
+    formData.append('file', newsletter.file);
+  }
+
+  const s3requestOptions = {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      Authorization: `Bearer ${token}`,
+    },
+    body: formData,
+  };
+  const s3response = await fetch(
+    url.resolve(BASE_URL, 'file/upload'),
+    s3requestOptions,
+  );
+  const s3body = await s3response.json();
+  if (s3body.status === 'ERROR') {
+    throw s3body;
+  }
+  const s3_url = s3body.data;
+
+  const requestOptions: RequestInit = {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      Authorization: `Bearer ${token}`,
+    },
+    body: JSON.stringify({
+      contact_id: newsletter.contact_id,
+      pdf_path: s3_url,
+      content: 'bogus content to satisfy the server',
+      type: 'letter',
+    }),
+  };
+  const response = await fetch(url.resolve(API_URL, 'letter'), requestOptions);
+  const body = await response.json();
+  if (s3body.status === 'ERROR') {
+    throw s3body;
+  }
+
+  const newsletterData: DirectMailLog = {
+    id: body.data.id,
+    fileLink: body.data.pdf_path,
+    delivered: body.data.delivered_count,
+    inTransit: body.data.in_transit_count,
+    returned: body.data.returned_count,
+    creationDate: new Date(body.data.created_at),
+    totalLettersCount: body.data.total_letter_count,
+    estimatedArrival: null,
+    contact_id: newsletter.contact_id,
+    status: null,
+  };
+  if (body.data.estimated_arrival) {
```

Comment:
cant we just declare it on the const instead of creating this if?
let's just do a ternary operator I think

Reply:
This code was for the old version with a log for individual letters so it's gone now